File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.java
package nerd.tuxmobil.fahrplan.congress.sharing;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;

import androidx.annotation.NonNull;

import nerd.tuxmobil.fahrplan.congress.MyApp;

public class SessionSharer {

    // String formattedSessions can be one or multiple sessions
    public static boolean shareSimple(@NonNull Context context, @NonNull String formattedSessions) {
        return share(context, formattedSessions, "text/plain");
    }

    public static boolean shareJson(@NonNull Context context, @NonNull String formattedSessions) {
        return share(context, formattedSessions, "text/json");
    }

    private static boolean share(@NonNull Context context, @NonNull String formattedSessions, @NonNull String mimeType) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_SEND);
        intent.putExtra(Intent.EXTRA_TEXT, formattedSessions);
        intent.setType(mimeType);
        PackageManager packageManager = context.getPackageManager();
        if (intent.resolveActivity(packageManager) == null) {
            MyApp.LogDebug(SessionSharer.class.getSimpleName(), "No activity to handle share intent.");
            return false;
        } else {
            context.startActivity(intent);
            return true;
        }
    }

}

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/sharing/SessionSharer.kt
package nerd.tuxmobil.fahrplan.congress.sharing

import android.content.ActivityNotFoundException
import android.content.Context
import androidx.core.app.ShareCompat

object SessionSharer {
    // String formattedSessions can be one or multiple sessions
    @JvmStatic
    fun shareSimple(context: Context, formattedSessions: String) {
        // Show system's share UI. It handles the case of no matching apps.
        val intent = ShareCompat.IntentBuilder(context)
            .setType("text/plain")
            .setText(formattedSessions)
            .createChooserIntent()
        context.startActivity(intent)
    }

    @JvmStatic
    fun shareJson(context: Context, formattedSessions: String): Boolean {
        val intent = ShareCompat.IntentBuilder(context)
            .setType("text/plain")
            .setText(formattedSessions)
            .intent

        // Don't use the system's share UI so Chaosflix can be selected as default app
        return try {
            context.startActivity(intent)
            true
        } catch (e: ActivityNotFoundException) {
            false
        }
    }
}
